File: Facade/Dart/HotelKeeperFacade.dart
import 'IRestaurant.dart';
import 'NonVegRestaurant.dart';
import 'VegNonBothRestaurant.dart';
import 'VegRestaurant.dart';

class HotelKeeperFacade {
  void getVegMenu() {
    VegRestaurant().getMenus()?.describe();
  }

  void getNonVegMenu() {
    NonVegRestaurant().getMenus()?.describe();
  }

  void getVegNonMenu() {
    VegNonBothRestaurant().getMenus()?.describe();
  }
}

File: Facade/Javascript/HotelKeeperFacade.js
import VegRestaurant from "./VegRestaurant.js";
import NonVegRestaurant from "./NonVegRestaurant.js";
import VegNonBothRestaurant from "./VegNonBothRestaurant.js";

class HotelKeeperFacade {
  getVegMenu() {
    const vegRestaurant = new VegRestaurant();
    vegRestaurant.getMenus().describe();
  }

  getNonVegMenu() {
    const nonVegRestaurant = new NonVegRestaurant();
    nonVegRestaurant.getMenus().describe();
  }

  getVegNonMenu() {
    const vegNonRestaurant = new VegNonBothRestaurant();
    vegNonRestaurant.getMenus().describe();
  }
}

export default HotelKeeperFacade;